### ISSUE TITLE
Samples

### ISSUE THREAD

**Issue by chevdor** [iu2frl/ICOM_MemoriesTools#1]
I read from Reddit that you have quite some radios. It would be great if you could dump some sample ICF file with the infos about the radio and the firmware version.
It may be convenient to code the model name and firmware version into the file name.

**Comment by ManoDaSilva** [comment]
I've created a few different versions from the cloning softwares that I have, each with one channel at 145.500 simplex, named "IU2FRL". 

[Profiles.zip](https://github.com/user-attachments/files/15658670/Profiles.zip)

I don't know how other series of radios work (ham or otherwise). However, for those ICF files, the first line (a number, such as 29790000) give out a unique identifier to which software it matches to. Afaik, each identifier will refer to a unique profile structure specific to this family of radios.  

- IC-FX06X_NXDN: Profile used for both mobiles and handsets, UHF or VHF, NXDN/Analog. Examples: IC-F5061, IC-F3061, 6061, etc. The software I'm using is version 5.30, which in my experience works with firmwares 5.6, 4.9 and possibly older ones too.
- IC-FX06X_dPMR: same thing, but in the dPMR/Analog variant (different firmwares, different cloning softwares). Software version is 2.20, tested on firmware 2.8d.
- IC-FX02X: for the 3020, 5010, 5020. I do not own those radios. Software version is 3.4. 
- IC-FX0: for the F70, F80 and F1700/F1800. Again, I do not own any of those models, software version is 2.2. 
- IC-FX400_NXDN: used for IC-F3400/6400/5400/4400. I have a few of those, the structure is quite different, and firmware versions are less clear. This is from software version 2.12.


Hoping it helps!

**Comment by iu2frl** [comment]
Here is an IC-2820 memory:

![image](https://github.com/iu2frl/ICOM_MemoriesTools/assets/61653175/31e1a646-68f3-4dda-85e8-eb1ebbb2d42e)
![image](https://github.com/iu2frl/ICOM_MemoriesTools/assets/61653175/60dd0d45-81e5-4968-bfc0-4166a66139c2)
![image](https://github.com/iu2frl/ICOM_MemoriesTools/assets/61653175/b1241dc6-6523-44e5-ad12-c1fbabdba089)
![image](https://github.com/iu2frl/ICOM_MemoriesTools/assets/61653175/c2f47d7b-fc4b-40bb-bafd-ff0d4526e6ee)

Channels are actually more but there's no easy way to export them in a user-friendly way

[2820.zip](https://github.com/user-attachments/files/15663763/2820.zip)

**Comment by iu2frl** [comment]
Here is an ID-51 plus

[0-99.csv](https://github.com/user-attachments/files/15664413/0-99.csv)
[100-199.csv](https://github.com/user-attachments/files/15664417/100-199.csv)
[200-299.csv](https://github.com/user-attachments/files/15664418/200-299.csv)
[300-399.csv](https://github.com/user-attachments/files/15664421/300-399.csv)
[400-499.csv](https://github.com/user-attachments/files/15664424/400-499.csv)
[ID-51_plus.zip](https://github.com/user-attachments/files/15664425/ID-51_plus.zip)

**Comment by iu2frl** [comment]
All radios are at the latest firmware available

**Comment by chevdor** [comment]
From what I have seen so far (already be mentioned), there is no real ICF standard. It seems to be a memory dump and thus it is very fragile and depends on the internal data structure. I guess this is due to laziness and/or to avoid having to do parsing in the radio. However, there is already some kinda parsing in and out of csv files. IMO csv is not the greatest format to serialize random data.

I think a good start would be to be able to spot the radio model and the firmware version from an ICF file itself.
From there, we could apply the right data structure to decode.

> However, for those ICF files, the first line (a number, such as 29790000) give out a unique identifier to which software it matches to. Afaik, each identifier will refer to a unique profile structure specific to this family of radios.

@ManoDaSilva great info. It is also to be expected that an ICF may work for a range of versions. I guess when they make a new firmware version, the version number in the ICF changes but the encoding may or may not. 
Without a proper encoding, it is however suprising and it would mean that the radio is able to load/parse "any" previous version of the ICF. That sounds at that point way less efficient that having a standard format so maybe they DO have a standard format, just unspotted yet.

**Comment by iu2frl** [comment]
You are right, ICF Is not a standard, as you can see It has different row and data length, that's why a common format should be found. U started with CSV just because it one of the easiest to handle and import export from chirp. Of course it does not consider additional info such as radio settings

**Comment by chevdor** [comment]
I am however a bit suprised that ICOM would not use some kind of standard format. It would make it hard for them to import and ICF that is a few firmware versions old... thinking....
